src/types/loans.ts

```diff
@@ -0,0 +1,38 @@
+import { MonetaryAmount } from "@interlay/monetary-js";
+import Big from "big.js";
+import { CurrencyExt } from "./currency";
+
+interface LoanReward{
+    currency: CurrencyExt;
+    apy: Big;
+}
+
+interface LoanPosition {
+    currency: CurrencyExt;
+    amount: MonetaryAmount<CurrencyExt>;
+    earnedReward: MonetaryAmount<CurrencyExt> | null; // null if rewards are not enabled.
+}
+
+interface LendPosition extends LoanPosition {
+    isCollateral: boolean;
+    earnedInterest: MonetaryAmount<CurrencyExt>;
+}
+
+type BorrowPosition = LoanPosition;
+
+interface LoanAsset {
+    currency: CurrencyExt;
+    lendApy: Big;
+    borrowApy: Big;
```

Comment:
There could be an asset that cannot be lend but could be borrowed. This Scenario happens for USDT on Compound. Same thing could happen the other way around. Can be lended, but not borrowed. For example LP's.

Reply:
@danielsimao Actually in our case, that is not possible. Check this thread where we discussed it: https://discord.com/channels/745259537707040778/1026832208612306974/1026836371081539705